datahub-web-react/src/app/entity/shared/containers/profile/header/EntityTabs.tsx

```diff
@@ -25,23 +25,22 @@ const Tab = styled(Tabs.TabPane)`
 `;
 
 export const EntityTabs = <T,>({ tabs, selectedTab }: Props) => {
-    const { entityData } = useEntityData();
+    const { entityData, loading } = useEntityData();
     const routeToTab = useRouteToTab();
     const baseEntity = useBaseEntity<T>();
 
+    const visibleTabs = tabs.filter((tab) => tab.display?.visible(entityData, baseEntity));
+    const enabledAndVisibleTabs = visibleTabs.filter((tab) => tab.display?.enabled(entityData, baseEntity));
+
     useEffect(() => {
-        if (!selectedTab) {
-            if (tabs[0]) {
-                routeToTab({ tabName: tabs[0].name, method: 'replace' });
-            }
+        if (!loading && !selectedTab && enabledAndVisibleTabs[0]) {
+            routeToTab({ tabName: enabledAndVisibleTabs[0].name, method: 'replace' });
         }
-    }, [tabs, selectedTab, routeToTab]);
-
-    const visibleTabs = tabs.filter((tab) => tab.display?.visible(entityData, baseEntity));
+    }, [loading, enabledAndVisibleTabs, selectedTab, routeToTab]);
 
     return (
         <UnborderedTabs
-            activeKey={selectedTab?.name}
+            activeKey={selectedTab?.name || ''}
```

Comment:
Scary, but if you say it works. The other thing we can do is only render the tabs if a tab has been selected:

{
   selectedTab && <UnborderedTabs...
}|

This would force that a Tab must first be selected in order for the Tabs to be rendered at all, which might make sense here (since there should always be SOME tab selected -- either the user selects, or we select for them)

Reply:
We should always render the tabs, because otherwise the page would somehow look a bit lost without "any content", additionally when the tabs are always rendered the user has already the possibility to select a tab until the data is loaded...technically there shouldn't be a reason to not render the tabs, or?

If we take a look on the url it makes also sense to have a state where no tab is selected:

/dashboard/urn:li:dashboard:(looker,baz)/Charts -> Charts tab
/dashboard/urn:li:dashboard:(looker,baz)/Documentation -> Documentation tab
/dashboard/urn:li:dashboard:(looker,baz)/ -> No tab?

This is how it looks like with the rendered tab component when no tab is selected:

![recording](https://user-images.githubusercontent.com/13187726/214424439-92ca3dab-3dc6-40f3-9345-61c0997d9f7e.gif)